GitCommands/Git/GitModule.cs

```diff
@@ -3057,12 +3052,12 @@ public string GetSelectedBranch(bool setDefaultIfEmpty)
 
             return result.ExitedSuccessfully
                 ? result.StandardOutput
-                : setDefaultIfEmpty ? DetachedHeadParser.DetachedBranch : string.Empty;
+                : emptyIfDetached ? string.Empty : DetachedHeadParser.DetachedBranch;
         }
 
         public string GetSelectedBranch()
```

Comment:
Please remove this method and add the optional parameter to the overload

Reply:
done.